docs/data/data-grid/column-groups/column-groups.md

```diff
@@ -1,18 +1,98 @@
 ---
-title: Data Grid - Column groups 🚧
+title: Data grid - Column groups
 ---
 
-# Data grid - Column groups 🚧
+# Data grid - Column groups
 
 <p class="description">Group your columns.</p>
 
+Grouping columns allows you to have multiple levels of columns in your header and to toggle column groups to show and hide additional columns.
+
+:::warning
+This feature is experimental, it needs to be explicitly activated using the `columnGrouping` experimental feature flag.
+
+```tsx
+<DataGrid experimentalFeatures={{ columnGrouping: true }} {...otherProps} />
+```
+
+:::
+
+## Define column groups
+
+You can define column groups with the `columnGroupingModel` prop.
+This prop receives an array of column groups.
+
+A column group is defined by at least two properties:
+
+- `groupId`: a string used to identify the group
+- `children`: an array containing the children of the group
+
+The `children` attribute can contain two types of objects:
+
+- leafs with type `{ field: string }`, which add the column with the corresponding `field` to this group.
+- other column groups which allows you to have nested groups.
+
+:::warning
+A column can only be associated with one group.
+:::
+
+```jsx
+<DataGrid
+  columnGroupingModel={[
+    {
+      groupId: 'internal data',
+      children: [{ field: 'id' }],
+    },
+    {
+      groupId: 'character',
+      children: [
+        {
+          groupId: 'naming',
+          children: [{ field: 'lastName' }, { field: 'firstName' }],
+        },
+        { field: 'age' },
+      ],
+    },
+  ]}
+/>
+```
+
+{{"demo": "BasicGroupingDemo.js", "bg": "inline"}}
+
+## Customize column group
+
+In addition to the required `groupId` and `children`, you can use the following optional properties to customize a column group:
+
+- `headerName`: the string displayed as the column's name (instead of `groupId`).
+- `description`: a text for the tooltip.
+- `renderHeaderGroup`: a function returning custom React component.
+
+{{"demo": "CustomizationDemo.js", "bg": "inline"}}
+
+## Column reordering [<span class="plan-pro"></span>](https://mui.com/store/items/mui-x-pro/)
+
+By default, the columns that are part of a group cannot be dragged to outside their group.
+You can customize this behavior on specific groups by setting `freeReordering: true` in a column group object.
+
+In the example below, the `Names` column group can be divided, but not other column groups
```

Comment:
It seems there's an issue with the reordering. Any idea why is this happening?

https://user-images.githubusercontent.com/550141/184096762-79b1cdb3-6df9-45c4-9ec1-11116d320437.mp4

Reply:
The first time when you go far right, it's not related to column grouping. It's because your mouse must be on top of a header when dropping. This only occurs when the total width is smaller than the container width (here is a demo to test)
https://codesandbox.io/s/columnorderingdisabledgrid-demo-mui-x-forked-m0muzg?file=/demo.ts

All the other behavior are normal. In the recording below, you can see:

1. Drag a column from the `freeReordering` group to the far right and far left.
2. The column from `no freeReordering` can not be dragged outside of its group (we can only inver the order od the "Internal" columns)

https://user-images.githubusercontent.com/45398769/184100322-8ade9ee5-5706-4c68-8453-3c53bd04ef68.mp4

Reply:
Alright. thanks for the clarification, I think it's worth opening a new issue for column reordering.